match2/commons/match_group_display_bracket.lua

```diff
@@ -1,738 +1,745 @@
+    local bracketResetMatch = match.bracketData.bracketResetMatchId
+        and props.matchesById[match.bracketData.bracketResetMatchId]
+    if bracketResetMatch then
+        match = MatchGroupUtil.mergeBracketResetMatch(match, bracketResetMatch)
     end
 
-    local skiproundConnector =
-        self.skipround > 0 and
-        html.create('div'):addClass('brkts-line-container'):node(
-            html.create('div'):addClass('brkts-line-horizontal-long')
-        ):node(
-            html.create('div'):addClass('brkts-line-spacer brkts-line-spacer-upper'):cssText(
-                'height:' .. self.padding .. 'px'
-            )
-        ):node(
-            html.create('div'):addClass('brkts-line-spacer brkts-line-spacer-lower'):cssText(
-                'height:' .. self.padding .. 'px'
-            )
-        ):node(html.create('div'):addClass('brkts-line-horizontal-long')) or
-        ''
-    skiproundConnector = tostring(skiproundConnector):rep(self.skipround)
-    local qualskipConnector =
-        self.qualskip > 0 and
-        html.create('div'):addClass('brkts-line-container'):node(
-            html.create('div'):addClass('brkts-line-horizontal-long')
-        ) or
-        ''
-    qualskipConnector = tostring(qualskipConnector):rep(self.qualskip)
-
-    -- qualified header
-    local qualText = _bracketConfig.qualifiedHeader
-    if (firsthead and ishead and self.qualwin and utils.misc.isEmpty(qualText)) then
-        local qualOptions = _getHeaderOptions('q')
-        qualText = qualOptions[1]
-        for i, option in ipairs(qualOptions) do
-            qualText = qualText .. tostring(html.create('div'):addClass('brkts-header-option'):node(option))
-        end
+    -- Current match
+    local matchNode = Bracket.Match({
+        MatchSummaryContainer = config.MatchSummaryContainer,
+        OpponentEntry = config.OpponentEntry,
+        match = match,
+        matchHasDetails = config.matchHasDetails,
+        opponentHeight = config.opponentHeight,
+    })
+        :css('margin-top', layout.matchMarginTop .. 'px')
+        :css('margin-bottom', config.matchMargin .. 'px')
+
+    -- Third place match
+    local thirdPlaceMatch = match.bracketData.thirdPlaceMatchId 
+        and props.matchesById[match.bracketData.thirdPlaceMatchId]
+    local thirdPlaceHeaderNode
+    local thirdPlaceMatchNode
+    if thirdPlaceMatch then
+        thirdPlaceHeaderNode = Bracket.MatchHeader({
+            header = '!tp',
+            height = config.headerHeight,
+        })
+            :css('margin-top', 20 + config.headerMargin .. 'px')
+            :css('margin-bottom', config.headerMargin .. 'px')
+        thirdPlaceMatchNode = Bracket.Match({
+            MatchSummaryContainer = config.MatchSummaryContainer,
+            OpponentEntry = config.OpponentEntry,
+            match = thirdPlaceMatch,
+            matchHasDetails = config.matchHasDetails,
+            opponentHeight = config.opponentHeight,
+        })
     end
```

Comment:
I feel these sections would benefit from more functions

Reply:
Most of this is passing props to `Match` and `MatchHeader`, so splitting it off to a separate component wouldn't help because you still have to pass props to it. Unless you mean passing the same props used in `NodeBody` to the separate component, but then it doesn't make sense for the 3rd place component to require a bunch of props that are unrelated to 3rd place.